### PR TITLE
ProfileServlet, Profile.jsp, web.xml, and ProfileServletTest

### PR DESCRIPTION
**Pull request by Millifiore**
- Created a Profile Servelet that connects to a JSP

- Created the JSP for the webpage.

- Mapped the servlet to the "Profile" page of the program.

- Created and ran a test for the ProfileServlet

### CODE REVIEW

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.io.IOException;
+
+public class ProfileServlet extends HttpServlet {
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+    }
+    /**
+     * This function fires when a user navigates to the Activity Feed page.
```
Comment (moustafaelsisy) [inline src/main/java/codeu/controller/ProfileServlet.java]: Woops; in their very subtle color, comments are easy to forget about x) The comment needs to be updated to reflect the Profile page

- [ ] Update comment

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--%>
+    <t:base>
+
+      <h1> Hello! ${username} </h1>
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/view/profile.jsp]: Sorry to bug you about this, but can we update this with the new template code? I will merge the PR for it into master now, and you can merge master into your branch to get the updates

File: src/main/webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -76,4 +76,14 @@
...
   </servlet-mapping>
 
+  <servlet>
+    <servlet-name>ProfileServlet</servlet-name>
+    <servlet-class>codeu.controller.ProfileServlet</servlet-class>
+  </servlet>
+
+  <servlet-mapping>
+    <servlet-name>ProfileServlet</servlet-name>
+    <url-pattern>Profile</url-pattern>
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/web.xml]: Usually URLs are lower case, so it seems that `/profile` will be a more suitable pattern that `Profile`

- [ ] Update URL accordingly

File: src/main/webapp/WEB-INF/view/profile.jsp (hunk 2 above)
Comment (moustafaelsisy) [reply src/main/webapp/WEB-INF/view/profile.jsp]: - [ ] Update to new templating style, and use scriptlet to retrieve username: `<%= request.getParameter("username") %>`

Comment (Millifiore) [reply src/main/webapp/WEB-INF/view/profile.jsp]: okay, thanks!

File: src/main/webapp/WEB-INF/web.xml [hunk 4]
```diff
@@ -76,4 +76,14 @@
...
   </servlet-mapping>
 
+  <servlet>
+    <servlet-name>ProfileServlet</servlet-name>
+    <servlet-class>codeu.controller.ProfileServlet</servlet-class>
+  </servlet>
+
+  <servlet-mapping>
+    <servlet-name>ProfileServlet</servlet-name>
+    <url-pattern>profile</url-pattern>
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/web.xml]: Can we add a `/` to the start of the pattern? It seems are servlet-mappings have them haha

Comment (Millifiore) [reply src/main/webapp/WEB-INF/web.xml]: for line 84?  I thought they were placed at the end, so I put mine on line 87.

Comment (moustafaelsisy) [reply src/main/webapp/WEB-INF/web.xml]: Oh no, I mean just replace `profile` with `/profile` on line 86 haha

Comment (Millifiore) [reply src/main/webapp/WEB-INF/web.xml]: oh okay!

**Review by ivannpe — APPROVED** [review]

**Review by moustafaelsisy — APPROVED** [review]
All looks good!

**Pull request merged**